### PR TITLE
SlotFill: remove explicit rerender from portal version

### PR DESCRIPTION
**Pull request by jsnajdr**
Simplifies the `SlotFill` implementation (the portal "bubbles virtually" version by removing explicit `rerender` API and its calls from `Fill`. After this patch the rerender happens automatically when the slot in the `slots` `observableMap` is changed with the `slots.set( name, ... )` call.

**How to test:**
All unit tests and e2e tests should pass.

### CODE REVIEW

File: packages/components/src/slot-fill/bubbles-virtually/fill.tsx [hunk 1]
```diff
@@ -20,18 +19,15 @@ import type { FillComponentProps } from '../types';
 export default function Fill( { name, children }: FillComponentProps ) {
 	const registry = useContext( SlotFillContext );
 	const slot = useObservableValue( registry.slots, name );
-	const [ , rerender ] = useReducer( () => [], [] );
-	const ref = useRef( { rerender } );
+	const instanceRef = useRef( {} );
```
Comment (jsnajdr) [inline packages/components/src/slot-fill/bubbles-virtually/fill.tsx:22]: The `rerender` API goes away and the fill now registers only an unique instance object. I'm renaming it from `ref` to `instance` when registering, something that @tyxla asked for in one of the previous PRs.

File: packages/components/src/slot-fill/bubbles-virtually/fill.tsx [hunk 2]
```diff
@@ -20,18 +19,15 @@ import type { FillComponentProps } from '../types';
...
-		// Some Slot implementations need to know if there're already fills
-		// registered so they can choose to render themselves or not.
-		const refValue = ref.current;
-		registry.registerFill( name, refValue );
-		return () => {
-			registry.unregisterFill( name, refValue );
-		};
+		const instance = instanceRef.current;
+		registry.registerFill( name, instance );
+		return () => registry.unregisterFill( name, instance );
```
Comment (jsnajdr) [inline packages/components/src/slot-fill/bubbles-virtually/fill.tsx:30]: This effect stays the same except renaming the `ref` variable to `instanceRef` etc.

File: packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.tsx [hunk 3]
```diff
@@ -23,13 +23,7 @@ function createSlotRegistry(): SlotFillBubblesVirtuallyContext {
...
 		fillProps
 	) => {
-		const slot = slots.get( name );
-
-		slots.set( name, {
-			...slot,
-			ref: ref || slot?.ref,
-			fillProps: fillProps || slot?.fillProps || {},
-		} );
+		slots.set( name, { ref, fillProps } );
```
Comment (jsnajdr) [inline packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.tsx:26]: A drive-by change removing redundant code. All the `registerSlot` calls always include both the `ref` and `fillProps` values. Inheriting them from the previous `slot` value has no effect.

File: packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.tsx [hunk 4]
```diff
@@ -66,12 +60,7 @@ function createSlotRegistry(): SlotFillBubblesVirtuallyContext {
 			return;
 		}
 
-		slot.fillProps = fillProps;
-		const slotFills = fills.get( name );
-		if ( slotFills ) {
-			// Force update fills.
-			slotFills.forEach( ( fill ) => fill.rerender() );
-		}
+		slots.set( name, { ref, fillProps } );
```
Comment (jsnajdr) [inline packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.tsx:63]: This is the key change in this PR! Instead of silently updating a field with `slots.get( name ).fillProps = fillProps`, we call `slots.set`. That sets a new value and triggers calling all `observableMap` listeners, leading to automatic rerendering of all fills.

File: packages/components/src/slot-fill/bubbles-virtually/slot.tsx [hunk 5]
```diff
@@ -54,11 +53,9 @@ function Slot(
...
 
 	useLayoutEffect( () => {
-		registerSlot( name, ref, fillPropsRef.current );
-		return () => {
-			unregisterSlot( name, ref );
-		};
-	}, [ registerSlot, unregisterSlot, name ] );
+		registry.registerSlot( name, ref, fillPropsRef.current );
+		return () => registry.unregisterSlot( name, ref );
+	}, [ registry, name ] );
```
Comment (jsnajdr) [inline packages/components/src/slot-fill/bubbles-virtually/slot.tsx:58]: A mere style change: don't destructure fields of `registry`. Simplifies the hook dependencies.

File: packages/components/src/slot-fill/bubbles-virtually/fill.tsx (hunk 1 above)
Comment (tyxla) [reply packages/components/src/slot-fill/bubbles-virtually/fill.tsx:22]: Thanks!

File: packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.tsx (hunk 3 above)
Comment (tyxla) [reply packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.tsx:26]: Makes sense 👍

File: packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.tsx (hunk 4 above)
Comment (tyxla) [reply packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.tsx:63]: Much clearer and more concise, looks great 🚀

**Review by tyxla — APPROVED** [review]
LGTM, nice simplification and cleanup 🚀

**Comment by Mamaduka** [comment]
Thank you, @jsnajdr!
